### DISCUSSION TITLE
Unable to access private gcr images even have right servie account defined to scanjob

### DISCUSSION THREAD

**Discussion by saurabh21316** [aquasecurity/trivy#4596] [Q&A]
### Question

Hello Team,

I would really appreciate if someone can help to resolve the private gcr docker images issue.
ERROR:
{"level":"error","ts":"2023-06-08T16:35:25Z","logger":"reconciler.scan job","msg":"Scan job container","job":"trivy-system/scan-vulnerabilityreport-8644d5d6fc","container":"limit-processer","status.reason":"Error","status.message":"2023-06-08T16:35:21.135Z\t\u001b[31mFATAL\u001b[0m\timage scan error: scan error: unable to initialize a scanner: unable to initialize a docker scanner: 4 errors occurred:\n\t* unable to inspect the image (us.gcr.io/abc-docker-image/limit-processer:3ef762665895ce7ca28ddd33d77904d113a5451c): Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?\n\t* unable to initialize Podman client: no podman socket found: stat podman/podman.sock: no such file or directory\n\t* containerd socket not found: /run/containerd/containerd.sock\n\t* GET https://us.gcr.io/v2/abc-docker-image/limit-processer/manifests/3ef762665895ce7ca28ddd33d77904d113a5451c: UNAUTHORIZED: You don't have the needed permissions to perform this operation, and you may have invalid credentials. To authenticate your request, follow the steps in: https://cloud.google.com/container-registry/docs/advanced-authentication\n\n\n","stacktrace":"github.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport/controller.(*ScanJobController).processFailedScanJob\n\t/home/runner/work/trivy-operator/trivy-operator/pkg/vulnerabilityreport/controller/scanjob.go:265\ngithub.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport/controller.(*ScanJobController).reconcileJobs.func1\n\t/home/runner/work/trivy-operator/trivy-operator/pkg/vulnerabilityreport/controller/scanjob.go:79\nsigs.k8s.io/controller-runtime/pkg/reconcile.Func.Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/reconcile/reconcile.go:103\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:118\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:314\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:265\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:226"}

Two thing I tried:
1.) Via apply GCP service account directly to trivy-operator instead of using default sa.
serviceAccountName: abcserver
serviceAccount: abcserver. 

2.) I also tried via apply annotation(gcp service account email id) to default trivy-operator sa.

None of it worked. I would highly appreciate if someone can point me to right direction.

Thanks!!

### Target

Container Image

### Scanner

Vulnerability

### Output Format

None

### Mode

Standalone

**Comment by saurabh21316** [comment]
Running in GKE via helm

**Comment by chen-keinan** [comment]
@saurabh21316 we have follow up on `trivy-operator` project
